superset-frontend/plugins/plugin-chart-echarts/src/Graph/EchartsGraph.tsx

```diff
@@ -143,7 +150,18 @@ export default function EchartsGraph({
           drillToDetail: drillToDetailFilters,
           crossFilter: getCrossFilterDataMask(node),
           drillBy: node && {
-            filters: [{ col: node.col, op: '==', val: node.name }],
+            filters: [
+              {
+                col: node.col,
+                op: '==',
+                val: node.name,
+                formattedVal: formatSeriesName(node.name, {
+                  timeFormatter: getTimeFormatter(node.name),
```

Comment:
The argument of `getTimeFormatter/getNumberFormatter` should be the time/number format selected by the user, available in `formData.dateFormat/formData.numberFormat` (or sth similar, not sure if the field name is the same for all charts)